ark/type/__tests__/defaults.test.ts

```diff
@@ -12,31 +15,48 @@ import { writeNonLiteralDefaultMessage } from "arktype/internal/parser/shift/ope
 contextualize(() => {
 	describe("parsing and traversal", () => {
 		it("base", () => {
-			const o = type({ foo: "string", bar: ["number", "=", 5] })
+			const o = type({
+				a: "string",
+				foo: "number = 5",
+				bar: ["number", "=", 5],
+				baz: ["number", "=", () => 5 as const]
+			})
+			type x = typeof o.t.baz
 
 			// ensure type ast displays is exactly as expected
-			attest(o.t).type.toString.snap("{ foo: string; bar: defaultsTo<5> }")
-			attest<{ foo: string; bar?: number }>(o.inferIn)
-			attest<{ foo: string; bar: number }>(o.infer)
+			attest(o.t).type.toString.snap(
+				"{ a: string; foo: defaultsTo<5>; bar: defaultsTo<5>; baz: defaultsTo<5> }"
+			)
+			attest<{ a: string; foo?: number; bar?: number; baz?: number }>(o.inferIn)
+			attest<{ a: string; foo: number; bar: number; baz: number }>(o.infer)
 
-			attest(o.json).snap({
-				required: [{ key: "foo", value: "string" }],
+			attest(o.omit("baz").json).snap({
+				required: [{ key: "a", value: "string" }],
 				optional: [
+					// WHY ARE THEY REVERSED?
```

Comment:
can you tell me why the order ok keys is reversed?

Reply:
No it's because all unordered conditions (branches of a union, keys of an object, etc.) are normalized so that types can be easily compared and we can reuse the cache.